assets/src/edit-story/output/test/story.js

```diff
@@ -220,6 +234,7 @@ describe('Story output', () => {
         pages: [
           {
             id: '123',
+            animations: [{ targets: ['123'], type: 'bounce', duration: 1000 }],
```

Comment:
![story-animation](https://user-images.githubusercontent.com/40646372/84101889-dbd35b00-a9c3-11ea-852c-628ec7ddd639.gif)

@dvoytenko  We're currently generating stories with animations using this structure:
```
<amp-story>
  <amp-story-page>
    <amp-animation id="anim1">
      <script>
        {
          "keyframes":{ "opacity":[0,1] }
        }
      </script>
    </amp-animation>

    <amp-story-animation>
      <script>
        {
          "animation": "anim1",
          "selector": "#element1",
          "duration": 1000,
        }
      </script>
    </amp-story-animation>

    <div id="element1">...</div>
  </amp-story-page>
</amp-story>
```

We're using the `amp-animation` tag to hold the `keyframes` for animations.  This way if we have multiple elements with the same animation, we don't have to repeat the `keyframes`.  The animated gif above is of a generate `amp-story` and as you can see the animations do work with this setup.  However, the tests are failing the `toBeValidAMP` assertions with the following error.
```
Tag 'amp-animation' is disallowed as child of tag 'amp-story-page'. Child tag must be one of ['amp-analytics',
'amp-pixel', 'amp-story-animation', 'amp-story-cta-layer', 'amp-story-grid-layer', 'amp-story-page-attachment'].
(see https://amp.dev/documentation/components/amp-story)
```

I tried changing `amp-animation` to `amp-story-animation` but then the animations break and the converted `amp-story-animation` throws the following error.

```
Element is not an animation: anim1
```

Is it possible to patch the `amp-story-animation` code in AMP so it allows for them to be used just to house `keyframes` like the `amp-animation` tag?  Or can `amp-animation` be added to the list of approved child tags for an `amp-story-page`?  Or do you feel there is a better third solution?

Reply:
Can we do

```html
<amp-story-animation>
      <script>
        {
          "keyframes":{ "opacity":[0,1] }
          "selector": "#element1",
          "duration": 1000,
        }
      </script>
</amp-story-animation>
```

for now?

That's how the examples do it:

https://github.com/ampproject/amphtml/blob/022bbcaa20896ea3050fd0f634851809104a742e/examples/amp-story/amp-story-animation.html

Reply:
We can, if that's the best approach for now I can update our animation rendering logic to include the keyframes within the `amp-story-animation` tags and then we can optimize it later if/when the original AMP animation method gets supported.

Reply:
Then let's do that 👍